### DISCUSSION TITLE
How to export the thermal conductivity to AuxVariables calculated by PorousFlowThermalConductivityIdeal object?

### DISCUSSION THREAD

**Discussion by lipxisi** [idaholab/moose#28811] [Q&A Modules: Porous Flow] [answered]
Holle

In PorousFlow module, I use PorousFlowThermalConductivityIdeal object to calculate the thermal conductivity  of the liquid - porous media system. I want to visualize the change in thermal conductivity, how do I do that?

Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

Since this is a material, there are parameters on the material object to output the properties.
Set "outputs = <name of the exodus output (called exodus by default iirc)>"
and you can also set the name of the properties to output

These will output the properties as auxiliary variables to Exodus

**Reply by lipxisi** [reply]
Hello

To calculate heat flux in my heat transfer calculations, I used the DiffusionFluxAux class in [AuxKernels]. But the 'diffusivity' of this class option cannot identify thermal conductivity 'PorousFlow_thermal_conductivity_ ' which is exported from PorousFlowThermalConductivityIdeal materials object using outputs = exodus. What should I do?

The following error occurs

''The following error occurred in the object "MOOSE Problem", of type "FEProblem".

Material property 'PorousFlow_thermal_conductivity_', requested by 'flux' is not defined on boundary fluid_solid_interface''

Thanks very much

**Reply by GiudGiud** [reply]
The output to exodus are auxiliary variables. DiffusionFluxAux expects the conductivity as a material property. You can use Debug/show_material_props to find the name of the mateiral property before its output to an auxvariable

**Reply by lipxisi** [reply]
Hello 

When I use Debug/show_material_props, the following information about flux and thermal conductivity is printed in the terminal:

`Consumed Material Properties:
    Object: Kernel::flux1
  Properties: PorousFlow_fluid_phase_density_nodal, PorousFlow_fluid_phase_density_qp, PorousFlow_grad_porepressure_qp, PorousFlow_mass_frac_nodal, PorousFlow_permeability_qp, PorousFlow_porepressure_nodal, PorousFlow_relative_permeability_nodal, PorousFlow_viscosity_nodal, dPorousFlow_fluid_phase_density_nodal_dvar, dPorousFlow_fluid_phase_density_qp_dvar, dPorousFlow_grad_porepressure_qp_dgradvar, dPorousFlow_grad_porepressure_qp_dvar, dPorousFlow_mass_frac_nodal_dvar, dPorousFlow_permeability_qp_dgradvar, dPorousFlow_permeability_qp_dvar, dPorousFlow_relative_permeability_nodal_dvar, dPorousFlow_viscosity_nodal_dvar

      Object: Kernel::heatconduct
  Properties: PorousFlow_grad_temperature_qp,  PorousFlow_thermal_conductivity_qp, dPorousFlow_grad_temperature_qp_dgradvar, dPorousFlow_grad_temperature_qp_dvar, dPorousFlow_thermal_conductivity_qp_dvar

      Object: Kernel::heatflux
  Properties: PorousFlow_fluid_phase_density_nodal, PorousFlow_fluid_phase_density_qp, PorousFlow_fluid_phase_enthalpy_nodal, PorousFlow_grad_porepressure_qp, PorousFlow_permeability_qp, PorousFlow_porepressure_nodal, PorousFlow_relative_permeability_nodal, PorousFlow_viscosity_nodal, dPorousFlow_fluid_phase_density_nodal_dvar, dPorousFlow_fluid_phase_density_qp_dvar, dPorousFlow_fluid_phase_enthalpy_nodal_dvar, dPorousFlow_grad_porepressure_qp_dgradvar, dPorousFlow_grad_porepressure_qp_dvar, dPorousFlow_permeability_qp_dgradvar, dPorousFlow_permeability_qp_dvar, dPorousFlow_relative_permeability_nodal_dvar, dPorousFlow_viscosity_nodal_dvar

     
      Object: MaterialBase::thermal_conductivity
  Properties: PorousFlow_saturation_qp, dPorousFlow_saturation_qp_dvar

      Object: MaterialBase::thermal_conductivity_face
  Properties: PorousFlow_saturation_qp, dPorousFlow_saturation_qp_dvar

      Object: MaterialBase::thermal_conductivity_neighbor
  Properties: PorousFlow_saturation_qp, dPorousFlow_saturation_qp_dvar`

I set the diffusivity option in the DiffusionFluxAux  to diffusivity = PorousFlow_thermal_conductivity_qp and terminal output following ERROR: 

*** ERROR ***
The following error occurred in the object "flux", of type "DiffusionFluxAux".
The requested AD material property 'PorousFlow_thermal_conductivity_qp' of type 'double'
is already retrieved or declared as a non-AD property of type 'libMesh::TensorValue<double>'.

How can I provide the correct thermal conductivity to the diffusivity option.

Thanks very much!

**Reply by GiudGiud** [reply]
DiffusionFluxAux expects a Real (single scalar) valued diffusivity. In porous flow the diffusivity is a tensor. 

I ll take a look but I don't think we have an auxkernel that handles a tensor in the framework

**Reply by cpgr** [reply]
You should be able to use MaterialRealTensorValueAux, but the better solution long-term would be to add conductivity to PorousFlowPropertyAux
